src/app/loadout-builder/generated-sets/GeneratedSetButtons.tsx

```diff
@@ -40,19 +48,73 @@ export default function GeneratedSetButtons({
     return dispatch(applyLoadout(store, loadout, true));
   };
 
+  const statsWithPlus5: number[] = [];
+
+  for (const [stat, value] of Object.entries(set.stats)) {
+    if (value % 10 > 4) {
+      statsWithPlus5.push(statHashes[stat]);
+    }
+  }
+
+  /**
+   * Adds as many of the half tier mods as it can as per the order of the filters stat values.
+   * Will show a notification if some mods couldn't be slotted.
+   */
+  const onQuickAddHalfTierMods = () => {
```

Comment:
May have missed the point of the reducer a bit - you're supposed to just send an action like "lock the half tier mods!" and then it calculates the new state given that input. The idea is to decouple the action from the state, or to "dispatch" a function on the state.

Reply:
Yeah just a result of doing things late at night. There is some stuff that will have to stay in this function as the mods are dependent on the armour set but I can move a bunch of it into the reducer. Mainly the stuff around only adding mods that we have space left for.

Reply:
I moved the show notification bit into the reducer as well...I dont know how I feel about that but the notification comes as a side effect of not having enough space for all the sent in mods. 

I could figure this out in the button handler but I pushed it all there for now. Let me know you thoughts and I am happy to move it back to the ui.